ConfigSpace/hyperparameters.py

```diff
@@ -836,11 +833,11 @@ def has_neighbors(self):
         """
         return len(self.sequence) > 1
 
-    def get_num_neighbors(self):
+    def get_num_neighbors(self, value):
```

Comment:
I'm afraid this won't work, as the base class does not have this parameters. The only usage of this method is in the method [get_one_exchange_neighborhood](https://github.com/Krxsy/ConfigSpace/blob/d7663059cbec3cd4494c875f2773afe7059935c4/ConfigSpace/util.py#L96). Since it's possible to obtain the hyperparameter value in that function, could you please update that function and add an additional argument to each hyperparameter?